api/serializers.py

```diff
@@ -1,74 +1,49 @@
 
-from .models import UserProfile, Video
+from .models import Profile, Video
 
-JWT_PAYLOAD_HANDLER = api_settings.JWT_PAYLOAD_HANDLER
-JWT_ENCODE_HANDLER = api_settings.JWT_ENCODE_HANDLER
 
-
-class UserSerializer(serializers.ModelSerializer):
+class ProfileSerializer(serializers.ModelSerializer):
     class Meta:
-        model = UserProfile
-        fields = ("first_name", "last_name", "phone_number", "age", "gender")
+        model = Profile
+        fields = "__all__"
 
 
-class UserRegistrationSerializer(serializers.ModelSerializer):
+class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ("username", "password")
-        extra_kwargs = {"password": {"write_only": True}}
+        fields = "__all__"
+
 
+class UserRegistrationSerializer(serializers.ModelSerializer):
     def create(self, validated_data):
         user = User.objects.create_user(**validated_data)
+        user.email = user.username
+        user.save()
 
         # A profile with default data is created here. We can add more as needed
         # in the profile model.
-        UserProfile.objects.create(
-            user=user,
-        )
+        user.profile = Profile(user.id)
+        user.profile.save()
         return user
 
-
-class UserLoginSerializer(serializers.Serializer):
-
-    username = serializers.CharField(max_length=255)
-    password = serializers.CharField(max_length=128, write_only=True)
-    token = serializers.CharField(max_length=255, read_only=True)
-
-    def validate(self, data):
-        username = data.get("username", None)
-        password = data.get("password", None)
-        user = authenticate(username=username, password=password)
-        if user is None:
-            raise serializers.ValidationError(
-                "A user with this email and password is not found."  # Not authenticated
-            )
-        try:
-            payload = JWT_PAYLOAD_HANDLER(user)
-            jwt_token = JWT_ENCODE_HANDLER(payload)
-            update_last_login(None, user)
-        except User.DoesNotExist:
-            raise serializers.ValidationError(
-                "User with given email and password does not exists"  # User not found
-            )
-        return {"username": user.username, "token": jwt_token}
+    class Meta:
+        model = User
+        fields = "__all__"
+        extra_kwargs = {"password": {"write_only": True}}
 
 
 class VideoSerializer(serializers.ModelSerializer):
     def create(self, validated_data):
         data = self.context.get("data")
         sender = self.context.get("user")
 
-        # receiver = User.objects.get(id=validated_data.pop('receiver'))
-        # Data is prepended by "data:video/mp4;base64,",
-        # need to remove those 22 characters
+        # Data is prepended by "data:video/mp4;base64,"
         # Encode converts from string to bytes
-        data_bytes = data[22:].encode("ascii")
+        data_bytes = data.encode("ascii")

```

Comment:
Still unclear as to why this is specifically being encoded with ascii. There is no benefit to encoding with ascii over utf-8. And you are decoding as utf-8 later.
```suggestion
        data_bytes = data.encode()

```